#! /bin/sh /usr/share/dpatch/dpatch-run
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: print the file and line number of the virtual host when
## DP: SSLCertificateFile is missing (#541607)

@DPATCH@
--- a/modules/ssl/ssl_engine_pphrase.c.dist	2006-07-23 13:11:58.000000000 +0200
+++ a/modules/ssl/ssl_engine_pphrase.c	2009-09-06 21:51:26.000000000 +0200
@@ -188,7 +188,8 @@
         if (sc->server->pks->cert_files[0] == NULL) {
             ap_log_error(APLOG_MARK, APLOG_ERR, 0, pServ,
                          "Server should be SSL-aware but has no certificate "
-                         "configured [Hint: SSLCertificateFile]");
+                         "configured [Hint: SSLCertificateFile] (%s:%d)",
+                         pServ->defn_name, pServ->defn_line_number);
             ssl_die();
         }
         algoCert = SSL_ALGO_UNKNOWN;
